### PR TITLE
Add Category Support

### PR DESCRIPTION
**Pull request by smithsps**
Took a somewhat simple plug-and-play approach

- Categories are formed from the unique set of categoryLabel fields in the item array.
- Categories cannot be selected
- Categories by default have no special format by default
- Categories are listed alphabetically

### CODE REVIEW

**Comment by lileesaraana** [comment]
You have a syntax error on line 42.

**Comment by lileesaraana** [comment]
Also might want to remove me from this repo ;)

File: playground/index.ts [hunk 1]
```diff
@@ -31,13 +31,15 @@ class AppComponent {
     'Laura',
     'Henery',
     'Texel',
+    'Simon',
```
Comment (lileesaraana) [inline playground/index.ts]: lol

Comment (lileesaraana) [reply playground/index.ts]: should probably change line 25

**Comment by smithsps** [comment]
I think you can unwatch it! I can't do anything otherwise though, I think. @adamk33n3r

**Comment by lileesaraana** [comment]
Ah yes, I will do that. :D

**Comment by adamk33n3r** [comment]
I can't test this out now but do arrow keys skip categories and do categories not change on hover?

File: src/mf-select.component.ts [hunk 2]
```diff
@@ -315,4 +323,32 @@ export class MfSelectComponent implements OnInit, AfterViewInit, OnChanges, OnDe
...
+        if (item[this.categoryLabel] === category) {
+          itemsWithCategories.push(item);
+        }
+      }
+    }
+
+    this.items = itemsWithCategories;
+  }
+
+  private isMfCategory(item: MfSelectItem): item is MfCategory {
```
[the remark this answers is not in the document]
Comment (smithsps) [reply src/mf-select.component.ts]: It's some typescript back magic called a User-Defined Type Guard. I think it's basically a Boolean in regular use

**Comment by adamk33n3r** [comment]
@lileesaraana I don't think you're "in" this repo. You must have just still had it watched.

**Comment by adamk33n3r** [comment]
> Probably not!

Then let's do that! :D

**Comment by adamk33n3r** [comment]
@smithsps ummmmm.....why did you just merge your own PR? And without requested changes nonetheless

**Comment by smithsps** [comment]
wtf happened?

**Pull request merged**